src/Meshes/interval.jl

```diff
@@ -167,34 +191,50 @@ function IntervalMesh(
     domain::IntervalDomain{CT},
     stretch::GeneralizedExponentialStretching{FT};
     nelems::Int,
+    reverse_mode::Bool = false,
 ) where {CT <: Geometry.Abstract1DPoint{FT}} where {FT}
     if nelems ≤ 1
         throw(ArgumentError("`nelems` must be ≥ 2"))
     end
+
     dz_bottom, dz_top = stretch.dz_bottom, stretch.dz_top
-    if !(dz_bottom ≤ dz_top)
+    if !(dz_bottom ≤ dz_top) && reverse_mode == false
         throw(ArgumentError("dz_bottom must be ≤ dz_top"))
     end
-    # surface coord height value
+
+    if !(dz_bottom ≥ dz_top) && reverse_mode == true
+        throw(ArgumentError("dz_top must be ≤ dz_bottom"))
+    end
+
+    # bottom coord height value, always min, for both atmos and land, since z-axis does not change
     z_bottom = Geometry.component(domain.coord_min, 1)
-    # top coord height value
+    # top coord height value, always max, for both atmos and land, since z-axis does not change
     z_top = Geometry.component(domain.coord_max, 1)
+    # but in case of reverse_mode, we temporarily swap them so that the following root solve algorithm does not need to change
+    if reverse_mode
+        z_bottom, z_top = Geometry.component(domain.coord_max, 1),
+        -Geometry.component(domain.coord_min, 1)
+    end
+    # for the same reason, in case of reverse_mode, we swap the roles of dz_bottom (largest for land, smallest for atmos) and dz_top (largest for atmos, smallest for land)
+    if reverse_mode
+        dz_top, dz_bottom = dz_bottom, dz_top
+    end
```

Comment:
```suggestion
    # but in case of reverse_mode, we temporarily swap them so that the following root solve algorithm does not need to change
    if reverse_mode
        z_bottom, z_top = Geometry.component(domain.coord_max, 1),
        -Geometry.component(domain.coord_min, 1)
        dz_top, dz_bottom = dz_bottom, dz_top
    end
```